### ISSUE TITLE
Wrong Mismatching Case Sensitivity

### ISSUE THREAD

**Issue by simPod** [slevomat/coding-standard#288]
`SomeClass.php`

```php
<?php

declare(strict_types=1);

use SomeClass2;

final class SomeClass
{
    /** @var SomeClass2 */
    private $someClass2;

    public function work() : string
    {
        return $this->someClass2::get();
    }
}
```

`SomeClass2.php`
```php
<?php

declare(strict_types=1);

abstract class SomeClass2
{
    abstract public static function get() : string;
}
```
Throws
> Case of reference name "someClass2" and use statement "SomeClass2" do not match

**Comment by iammattcoleman** [comment]
I am also getting false positives for `UnusedUses.MismatchingCaseSensitivity` after running `composer update` just now:
```php
<?php
use Foo\Bar;
use Foo\BarInterface;

/**
 * @param BarInterface $bar BAR to to create with.
 */
function foo(BarInterface $bar)
{
}
```

This test passes on 4.4.3 but fails on all newer versions. The failure state is not the same from version to version...

On 4.4.3:
```
6 | ERROR | [ ] Case of reference name bar and use statement Bar do not match. (SlevomatCodingStandard.Namespaces.UnusedUses.MismatchingCaseSensitivity)
```

On 4.4.5:
```
 2 | ERROR | [x] Type Foo\Bar is not used in this file. (SlevomatCodingStandard.Namespaces.UnusedUses.UnusedUse)
```

On 4.4.6 (same as 4.4.4):
```
 6 | ERROR | [ ] Case of reference name "BAR" and use statement "Bar" do not match. (SlevomatCodingStandard.Namespaces.UnusedUses.MismatchingCaseSensitivity)
```

**Comment by kukulich** [comment]
@iammattcoleman Your bug is probably not fixable...

**Comment by kukulich** [comment]
@iammattcoleman Hmm, it should be, I’ll try :)

**Comment by iammattcoleman** [comment]
@kukulich Thanks for fixing this!
